### DISCUSSION TITLE
Simplify cover entities which support reporting or setting position

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#1011] [Entity Models]
## Proposal

1. Cover entities which support reporting position should no longer implement the `is_closed` property, but must instead set the new boolean property `reports_position`. The base entity will decide if the cover is closed or open based on the cover's position and ignore the `is_closed` property for cover entities which set `reports_position` to `True`.

2. Cover entities which support setting position or tilt position do no longer need to implement the `close` or `open` service handlers. Instead, the base entity will call the `set_position` or `set_tilt_position` with position set to either `0` or `100`.

The upcoming `valve` integration, implemented by PR https://github.com/home-assistant/core/pull/102184, can be used as a reference.

### Backwards compatibility

If the property `reports_position` is not set, cover entities will during a deprecation period behave like before. After the deprecation period has ended, it's an error to not set the `reports_position` property.

## Rationale

There's a request to be able to "invert" a cover. This is not possible if cover entities which support reporting position do their own implementation of the `is_closed` property.

**Comment by frenck** [comment]
> If the property reports_position is not set, cover entities will during a deprecation period behave like before. After the deprecation period has ended, it's an error to not set the reports_position property.

And after that a full removal of it?

**Reply by emontnemery** [reply]
No, we still need the `reports_position` property to know if the entity reports an absolute position or not.
